src/sentry/api/endpoints/project_details.py

```diff
@@ -512,3 +519,55 @@ def delete(self, request, project):
             )
 
         return Response(status=204)
+
+    def transfer_project(self, request, project):
```

Comment:
i think this should probably just be it's own endpoint because this endpoint already is pretty complicated. Then you can also just adjust the permissions of the whole endpoint accordingly without having to shove this in the delete method